asreview/logging/utils.py

```diff
@@ -3,18 +3,18 @@
 
 def get_logger_class(fp):
```

Comment:
I think we can drop the class `Logger`. The functionality can be combined with this function. 

```
logger = get_logger_class(fp).from_file(fp)
```

This might require some work because of the class attribute `Logger.logger`. But this makes the entire API consistent.

Reply:
Well, except it wouldn't work properly as a context manager or would it? Maybe it makes sense to make get_logger_class private?

Reply:
It does look kind of ugly with two fp's, no?

Reply:
I had a more detailed look at what is going on. Hopefully, I can clarify it a bit. 

I think there should be a function `get_logger(fp)` that initializes `_get_logger_class(fp)`. This function outputs an initiated object that inherits from `BaseLogger`. The class `Logger` can get removed. Context managing is done in `BaseLogger`. 

Throughout the code library, we can replace  

``` python
with Logger.from_file(log_file) as logger:
     if not logger.is_empty():
```
 with
``` python
with get_logger(log_file) as logger:
     if not logger.is_empty():
```

This syntax is in line with other Python libraries. Please have a look at `get_logger` in tensorflow and `open` in base. 

The last one is a good example of the prefered syntax. One can call `with open(fp) as f: `. This returns an `_io.TextIOWrapper` object named `f`. What we do: We call something like `_io.TextIOWrapper.read(fp)` instead of using our util `get_logger`. 

And while we're at it, think about renaming `get_logger()` into `open_logger()`. Completely in line with `open()`

Reply:
I'm still not exactly sure what you meant, but check the new implementation, I think that should work fine, right?